features/frameworks/utest/source/utest_serial.cpp

```diff
@@ -0,0 +1,62 @@
+/****************************************************************************
+ * Copyright (c) 2019, ARM Limited, All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ****************************************************************************
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <cstdarg>
+
+#include "platform/mbed_retarget.h"
+
+
+#define STRING_STACK_LIMIT    120
+
+
+static void utest_puts(const char *str)
+{
+    write(STDERR_FILENO, str, strlen(str));
+}
+
+static int utest_vprintf(const char *format, std::va_list arg)
+{
+    // ARMCC microlib does not properly handle a size of 0.
+    // As a workaround supply a dummy buffer with a size of 1.
+    char dummy_buf[1];
+    int len = vsnprintf(dummy_buf, sizeof(dummy_buf), format, arg);
+    if (len < STRING_STACK_LIMIT) {
+        char temp[STRING_STACK_LIMIT];
+        vsprintf(temp, format, arg);
+        utest_puts(temp);
```

Comment:
If this is "utest for greentea", shouldn't it call `greentea_write_string`? Can't quite decide whether this decoupling from Greentea is good or bad.

I think I'd only decouple if it would totally decouple this utest stuff - but if you are assuming Greentea elsewhere, this should "play nice" and use `greentea_write_string`.

Reply:
Please review [this](https://github.com/ARMmbed/mbed-os/compare/10b2d1c8aa95b39231cb96777f816895ce9538bb..72e390ca702f6528c2c52c576808d7a4decbbfd5).